crates/sui/src/keytool.rs

```diff
@@ -343,32 +343,38 @@ impl KeyToolCommand {
                 key_scheme,
                 derivation_path,
             } => {
-                // check if it is a private key or a mnemonic
-                if input_string.starts_with("0x") && input_string.len() == 66 {
-                    let base64 = convert_string_to_base64(input_string)?;
-                    let skp = SuiKeyPair::decode_base64(&base64);
-                    if let Ok(skp) = skp {
-                        keystore.add_key(skp)?;
-                        println!("Private key imported successfully.")
-                    } else {
-                        return Err(anyhow!(
-                            "Cannot decode base64 from private key. Importing failed."
-                        ));
+                // check if input is a private key -- should start with 0x
+                if input_string.starts_with("0x") {
+                    let bytes = Hex::decode(&input_string)
+                        .map_err(|_| anyhow!("Private key is malformed. Importing failed"))?;
+                    match key_scheme {
+                        SignatureScheme::ED25519 => {
+                            let kp = Ed25519KeyPair::from_bytes(&bytes).map_err(|_| anyhow!("Cannot decode ed25519 keypair from the private key. Import failed."))?;
+                            let skp = SuiKeyPair::Ed25519(kp);
+                            let address = Into::<SuiAddress>::into(&skp.public()).to_string();
```

Comment:
Nit: Rust idioms for using `into` -- and then can you rely on the `println` to do the string conversion for you? (It will use the `Display` impl for `SuiAddress` which I'm hoping will do the right thing).

```suggestion
                            let address: SuiAddress = skp.public().into();
```

Reply:
Thanks for the suggestion - I used existing code for this, as otherwise the compiler complains about the into() method
```
the trait bound `sui_types::crypto::PublicKey: Into<_>` is not satisfied
required for `sui_types::crypto::PublicKey` to implement `Into<SuiAddress>`
```